zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java

```diff
@@ -879,8 +887,12 @@ private void uploadAdapterFile() {
             HRawDocument rawDocument = new HRawDocument();
             rawDocument.setDocument(document);
             rawDocument.setContentHash(new String(Hex.encodeHex(md5hash)));
-            rawDocument.setType(DocumentType.typeFor(translationFileServiceImpl
-                    .extractExtension(fileName)));
+            if(docType.isPresent()) {
+                rawDocument.setType(DocumentType.getByName(docType.get()));
+            } else {
+                rawDocument.setType(DocumentType.fromSourceExtension(
+                    FileUtil.extractExtension(fileName)).iterator().next());
```

Comment:
- [ ] Why does this have `.iterator().next()` on it? Is `extractExtension()` returning a collection instead of just a single extension? I would only expect to find one extension on a filename.